prov/rxd/src/rxd_atomic.c

```diff
@@ -63,29 +64,38 @@ static struct rxd_x_entry *rxd_tx_entry_init_atomic(struct rxd_ep *ep, fi_addr_t
 		memcpy(&tx_entry->res_iov[0], res_iov, sizeof(*res_iov) * res_count);
 	}
 
+	max_inline = rxd_domain->max_inline_msg;
 	if (rma_count > 1 || tx_entry->cq_entry.flags & FI_READ) {
 		rxd_init_base_hdr(ep, &ptr, tx_entry);
 		rxd_init_sar_hdr(&ptr, tx_entry, rma_count);
+		max_inline -= sizeof(struct rxd_sar_hdr);
 	} else {
 		tx_entry->flags |= RXD_INLINE;
 		tx_entry->num_segs = 1;
 		rxd_init_base_hdr(ep, &ptr, tx_entry);
 	}
 
-	if (tx_entry->flags & RXD_REMOTE_CQ_DATA)
+	if (tx_entry->flags & RXD_REMOTE_CQ_DATA) {
+		max_inline -= sizeof(struct rxd_data_hdr);
 		rxd_init_data_hdr(&ptr, tx_entry);
+	}
 
 	rxd_init_rma_hdr(&ptr, rma_iov, rma_count);
+	max_inline -= sizeof(struct rxd_rma_hdr);
 	rxd_init_atom_hdr(&ptr, datatype, atomic_op);
+	max_inline -= sizeof(struct rxd_atom_hdr);
```

Comment:
Compiler will probably do this, but the max_inline adjustments can be combined to save one write.

Reply:
Will update